### PR TITLE
Feature/answers for coding questions

### PR DESCRIPTION
**Pull request by lee2000n**

### CODE REVIEW

File: src/api/posts.js [hunk 1]
```diff
@@ -41,4 +42,58 @@
...
+            },
+            
+          },
+        },
+      ],
+    });
+
+    const postIds = users_post.map(item => item.postId);
+
+    const postList = Post.findAll({
```
Comment (hatchy-reviewer) [inline src/api/posts.js:72]: You've done a good job using a bulk query to both fetch and sort the posts by multiple users

File: src/api/posts.js [hunk 2]
```diff
@@ -41,4 +42,58 @@
...
+      include: [
+        {
+          where: {
+            id: {
+              [opIn]:postIds
+            },
+            order: [
+              ['id', 'ASC'],
+              ['popularity', 'ASC'],
+          ],
```
Comment (hatchy-reviewer) [inline src/api/posts.js:82]: You hardcoded the value for sortBy and direction. you should use the parameters supplied with the request

File: src/api/updateController.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
+const db = require('../db/models');
+
+
+const UpdatePost = db.Post;
+exports.update = (req, res) => {
+    const { postId } = req.params;
+  const err="an error message";
+    UpdatePost.update(req.body, {
```
Comment (hatchy-reviewer) [inline src/api/updateController.js:8]: You should properly validate the req.body before updating the post in the database

File: src/api/Part3.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
+1. 
+// What database changes would be required to the starter code to allow for different roles for authors of a blog post?
+in database,
+Create a roles table ('Admin', 'User', 'Guest');
+Create a permissions table ('Create', 'Read','Write','Delete','Deny');
+Create a junction table with all three tables as sources.
```
[the remark this answers is not in the document]
Comment (lee2000n) [reply src/api/Part3.md:6]: it is done answering.

File: src/api/Part3.md [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+
+2. How would you have to change the PATCH route given your answer above to handle roles?
+
+it can import the middlewares.js file and have 'next' in the patch method
```
[the remark this answers is not in the document]
Comment (lee2000n) [reply src/api/Part3.md:13]: it is done updating.

File: src/api/posts.js [hunk 6]
```diff
@@ -41,4 +42,58 @@ router.post('/', async (req, res, next) => {
   }
 });
 
+//Part 1: Fetching Blog Posts
+
+router.get('/api/posts', async (req, res, next) => {
```
[the remark this answers is not in the document]
Comment (lee2000n) [reply src/api/posts.js]: it is done updating.

File: src/api/posts.js [hunk 7]
```diff
@@ -41,4 +42,58 @@
...
+  try {
+    const error = new Error("An error message");
+    const {authorIds} = req.query;
+    if(!authorIds){
+      return res.status(400).send({
+        message: error
+     });
+    }
+   const id_list = authorIds.split(',');
+    const users_post =  UserPost.findAll({
```
[the remark this answers is not in the document]
Comment (lee2000n) [reply src/api/posts.js]: it is done updating.

File: src/api/posts.js (hunk 1 above)
Comment (lee2000n) [reply src/api/posts.js:71]: thank you for your praise.^_^

File: src/api/posts.js (hunk 2 above)
Comment (lee2000n) [reply src/api/posts.js:81]: it is done updating.

File: src/api/posts.js [hunk 8]
```diff
@@ -41,4 +42,58 @@
...
+      ],
+    });
+
+
+    res.json({ post:postList });
+  } catch (error) {
+    next(error);
+  }
+//part 2 Updating a Blog Post
+router.put(`/api/posts/${UserPost.postId}`, update);
```
[the remark this answers is not in the document]
Comment (lee2000n) [reply src/api/posts.js]: i have created the patch way to replace the put.

**Comment by lee2000n** [comment]
> Hi, I'm Ikechukwu from Hatchways and I'll be reviewing your code today. You can re-submit your assessment by doing the following:
> 
> * respond to any comments on GitHub and address them by pushing more commits to this branch in this pull request
> * when you are ready for us to look at the pull request again, go back to this page and click the "Re-submit" button for this assessment. If you don't do this step, we won't know that you want us to look at the pull request again
> 
> By re-submitting, it can drastically improve your chances to move forward to the next steps in the interview process. Here is some general feedback about the correctness of your solution:
> 
> Your solution does not properly map the route handlers to the proper routes. You do not protect the get route from unauthorized users Your solution does not have a PATCH route Your solution does not properly validate the request parameters to ensure that they match the route specifications for both routes e.g in the GET route, you do not ensure that the authorIds parameter is a comma-separated list of integers

Hi Ikechukwu, i have done the changes, but as im not good at coding the unit test, i can not resubmit the merge request....., may i know if you could check the changes i made from the existing pull request ? thank you

**Pull request closed without merging**